File: Backend API/CleaningHouse_API/Models/Customers/Report.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using CleaningHouse_API.Models.Common;
using CleaningHouse_API.Models.Companies;

namespace CleaningHouse_API.Models.Customers;

[Table("Reports")]
public class Report
{
    [Key]
    public int Id { get; set; }

    [Required]
    public int UserId { get; set; }

    [Required]
    public ReportTargetType TargetType { get; set; }

    public int? WorkerId { get; set; }

    public int? CompanyId { get; set; }

    [Required]
    [MaxLength(2000)]
    public string Description { get; set; } = string.Empty;

    public ReportStatus Status { get; set; } = ReportStatus.Pending;

    [MaxLength(2000)]
    public string? AdminNotes { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public DateTime? UpdatedAt { get; set; }

    [ForeignKey(nameof(UserId))]
    public virtual AppUser? AppUser { get; set; }

    [ForeignKey(nameof(WorkerId))]
    public virtual Worker? Worker { get; set; }

    [ForeignKey(nameof(CompanyId))]
    public virtual Company? Company { get; set; }
}

File: App_Clint/lib/features/reports/domain/entities/report.dart
import 'package:equatable/equatable.dart';

enum ReportTargetType {
  worker,
  company;

  int get apiValue => switch (this) {
        ReportTargetType.worker => 1,
        ReportTargetType.company => 2,
      };

  static ReportTargetType fromApi(int value) => switch (value) {
        1 => ReportTargetType.worker,
        2 => ReportTargetType.company,
        _ => ReportTargetType.worker,
      };
}

enum ReportStatus {
  pending,
  underReview,
  resolved,
  dismissed;

  static ReportStatus fromApi(int value) => switch (value) {
        0 => ReportStatus.pending,
        1 => ReportStatus.underReview,
        2 => ReportStatus.resolved,
        3 => ReportStatus.dismissed,
        _ => ReportStatus.pending,
      };
}

/// Customer report entity — [adminNotes] is never exposed to UI.
class Report extends Equatable {
  const Report({
    required this.id,
    required this.userId,
    this.userName,
    required this.targetType,
    this.targetTypeName,
    this.workerId,
    this.workerName,
    this.companyId,
    this.companyName,
    required this.description,
    required this.status,
    required this.statusName,
    required this.createdAt,
    this.updatedAt,
  });

  final int id;
  final int userId;
  final String? userName;
  final ReportTargetType targetType;
  final String? targetTypeName;
  final int? workerId;
  final String? workerName;
  final int? companyId;
  final String? companyName;
  final String description;
  final ReportStatus status;
  final String statusName;
  final DateTime createdAt;
  final DateTime? updatedAt;

  String get targetDisplayName {
    if (targetType == ReportTargetType.worker) {
      return workerName ?? targetTypeName ?? '';
    }
    return companyName ?? targetTypeName ?? '';
  }

  @override
  List<Object?> get props => [
        id,
        userId,
        userName,
        targetType,
        targetTypeName,
        workerId,
        workerName,
        companyId,
        companyName,
        description,
        status,
        statusName,
        createdAt,
        updatedAt,
      ];
}
